### PR TITLE
style: enable `MissingSwitchDefault`  in checkstyle

### PR DESCRIPTION
**Pull request by BamaCharanChhandogi**
related #5119 

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/misc/Sort012D.java [hunk 1]
```diff
@@ -51,11 +56,16 @@ public static void sort012(int[] a) {
                 h--;
                 break;
             }
+            default: {
+                System.out.println("Unexpected value: " + a[mid]);
+                mid++;
+                break;
```
Comment (vil02) [inline src/main/java/com/thealgorithms/misc/Sort012D.java:62]:
```suggestion
                throw new IllegalArgumentException("Unexpected value: " + a[mid]);
```

File: src/main/java/com/thealgorithms/sorts/DNFSort.java [hunk 2]
```diff
@@ -31,6 +31,11 @@ static void sort012(int[] a, int arr_size) {
                 high--;
                 break;
             }
+            default: {
+                System.out.println("Unexpected value: " + a[mid]);
+                mid++;
+                break;
```
Comment (vil02) [inline src/main/java/com/thealgorithms/sorts/DNFSort.java:37]:
```suggestion
                throw new IllegalArgumentException("Unexpected value: " + a[mid]);
```

File: src/main/java/com/thealgorithms/strings/ValidParentheses.java [hunk 3]
```diff
@@ -26,6 +29,10 @@ public static boolean isValid(String s) {
             case ']':
                 if (head == 0 || stack[--head] != '[') return false;
                 break;
+            default:
+                // Optional: Handle unexpected characters if needed
+                System.out.println("Unexpected character: " + c);
+                return false;
```
Comment (vil02) [inline src/main/java/com/thealgorithms/strings/ValidParentheses.java:35]:
```suggestion
                throw new IllegalArgumentException("Unexpected character: " + c);
```

File: src/main/java/com/thealgorithms/stacks/PostfixToInfix.java
Comment (vil02) [inline src/main/java/com/thealgorithms/stacks/PostfixToInfix.java:1]: Please revert all of the non-necessary changes and create a new PR with them.

**Review by vil02 — CHANGES_REQUESTED** [review]

**Comment by BamaCharanChhandogi** [comment]
ok

**Pull request closed without merging**